app/src/main/java/com/github/multimatum_team/multimatum/ReminderBroadcastReceiver.kt

```diff
@@ -17,38 +20,45 @@ class ReminderBroadcastReceiver : BroadcastReceiver() {
     @Inject
     lateinit var deadlineRepository: DeadlineRepository
 
+    private val notifEnableKey =
+        "com.github.multimatum_team.multimatum.activity.MainSettingsActivity.NotifEnabled"
+
     /**
      * When an intent of this broadcast is started, it will create a notification with the right parameters and launch it
      */
     override fun onReceive(context: Context, intent: Intent?) {
-        val channelId = "remindersChannel"
-
-        //retrieving some parameters for the notification
-        val title = intent!!.getStringExtra("title")
-        val content = intent!!.getStringExtra("description")
-        val notificationId = intent!!.getStringExtra("id")
-
-
-        val intent2 = Intent(context, MainActivity::class.java)
-        intent2!!.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
-        //this is the intent where the user will be send when clicking on the notification
-        val pendingIntent =
-            PendingIntent.getActivity(context, 0, intent2, PendingIntent.FLAG_IMMUTABLE)
-
-        //Builder of the notification
-        val notificationBuilder = NotificationCompat.Builder(context!!, channelId)
-            .setSmallIcon(R.drawable.ic_launcher_foreground)
-            .setContentTitle(title)
-            .setContentText(content)
-            .setPriority(NotificationCompat.PRIORITY_DEFAULT)
-            .setContentIntent(pendingIntent)
-            .setAutoCancel(true)
-            .setCategory(NotificationCompat.CATEGORY_REMINDER)
-            .setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
-
-        //send the notification
-        val notificationManager =
-            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
-        notificationManager.notify(notificationId.hashCode(), notificationBuilder.build())
+        Log.i("sp notif_enable", PreferenceManager.getDefaultSharedPreferences(context).getBoolean(notifEnableKey, true).toString())
+        if(PreferenceManager.getDefaultSharedPreferences(context).getBoolean(notifEnableKey, true)) {
```

Comment:
```suggestion
        val notifEnabled = PreferenceManager.getDefaultSharedPreferences(context).getBoolean(notifEnableKey, true)
        Log.i("sp notif_enable", notifEnabled.toString())
        if(notifEnabled) {
```

Also, is the `SharedPreferences` you use the same instance as the one that is provided by `DependenciesProvider` and injected in several classes? If yes wouldn't it be better to inject it here too (for testing, and clarity, and to avoid problems if we change the implementation of the method in `DependenciesProvider` that provides it? If no, does disabling the notifications in the settings activity really prevent notifications to be sent?

Reply:
sure